src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -407,24 +408,37 @@ def insertContainerRules(self):
         # Create the subscription objects and add them to the list
         # The list takes care of the sorting internally
         for subInfo in unsubscribedDatasets:
-            rse = subInfo['site']
+            kwargs = dict(ask_approval=False, activity="Production Output",
+                          account=self.rucioAcct, grouping="ALL",
+                          comment="WMAgent automatic container rule", meta=self.metaData)
+            rse = subInfo['site'].replace("_MSS", "_Tape")
             container = subInfo['path']
             if not self._isContainerTierAllowed(container):
                 logging.debug("Component configured to skip container rule for: %s", container)
                 continue
             logging.info("Creating container rule for %s against RSE %s", container, rse)
 
             rseName = "%s_Test" % rse if self.testRSEs else rse
-            # ALL = replicates all files to the same RSE
-            resp = self.rucio.createReplicationRule(container, rseExpression="rse=%s" % rseName,
-                                                    account=self.rucioAcct, grouping="ALL",
-                                                    comment="WMAgent automatic container rule",
-                                                    meta=self.metaData)
+            try:
+                # ALL = replicates all files to the same RSE
+                resp = self.rucio.createReplicationRule(container,
+                                                        rseExpression="rse=%s" % rseName, **kwargs)
+            except WMRucioException as exc:
+                msg = "Failed to create container rule for (retrying with approval): %s" % container
+                logging.warning(msg)
+                kwargs["ask_approval"] = True
+                try:
+                    resp = self.rucio.createReplicationRule(container,
+                                                            rseExpression="rse=%s" % rseName, **kwargs)
+                except Exception as exc:
+                    msg = "Failed once again to create container rule for: %s" % container
```

Comment:
Maybe in this message you should say that exactly the (retry with approval) failed. I know one can figure it out from the sequence the messages enter the log but having it stressed will make it easier to connect this failure with the one from above while looking at the logs.

Reply:
Or one can read the "Failed once again", which means there was a previous failure ;)

Reply:
@amaltaro - I said 'Maybe'.  It is a matter of taste. It is fine as it is too. I already approved it :)